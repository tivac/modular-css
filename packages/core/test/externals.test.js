"use strict";

var dedent = require("dedent"),
    namer  = require("test-utils/namer.js"),
    
    Processor = require("../processor.js");

describe("/processor.js", () => {
    describe("externals", () => {
        var processor;
        
        beforeEach(() => {
            processor = new Processor({
                namer
            });
        });
        
        it("should fail if not a valid composition reference", () =>
            processor.string(
                "./invalid-external.css",
                dedent(`
                    :external(some garbage here) { }
                `)
            )
            .catch((error) =>
                expect(error.message).toMatch(`SyntaxError: Expected`)
            )
        );

        it("should fail if not referencing another file", () =>
            processor.string(
                "./invalid-external.css",
                dedent(`
                    :external(a) { }
                `)
            )
            .catch((error) =>
                expect(error.message).toMatch(`externals must be from another file`)
            )
        );

        it("should fail on bad class references", () =>
            processor.file(require.resolve("./specimens/externals-invalid.css"))
            .catch((error) =>
                expect(error.message).toMatch(`Invalid external reference: nopenopenope`)
            )
        );
        
        it("should support overriding external values", () =>
            processor.file(
                "./packages/core/test/specimens/externals.css"
            )
            .then(() => processor.output())
            .then((result) =>
                expect(result.css).toMatchSnapshot()
            )
        );
    });
});
