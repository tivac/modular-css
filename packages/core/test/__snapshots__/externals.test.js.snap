// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/processor.js externals should support overriding external values 1`] = `
"/* packages/core/test/specimens/folder/folder.css */
.folder { margin: 2px; }

/* packages/core/test/specimens/simple.css */
.wooga { color: red; }

/* packages/core/test/specimens/local.css */
.booga { background: green; }

/* packages/core/test/specimens/externals.css */
.a .wooga {
    color: green;
}
.b:hover .booga {
    background: blue;
}
"
`;
